Cython/Shadow.py

```diff
@@ -233,6 +233,15 @@ def __exit__(self, exc_class, exc, tb):
 del _nogil
 
 
+class critical_section:
+    def __init__(self, *args):
+        pass
+    def __enter__(self):
+        pass
+    def __exit__(self, exc_class, exc, tb):
+        return False
```

Comment:
This doesn't make it work as a decorator. Why not use the existing generic implementations?